app/scripts/controllers/builds.js

```diff
@@ -79,22 +77,15 @@ angular.module('openshiftConsole')
         }
 
         function associateBuildsToBuildConfig() {
-          $scope.buildsByBuildConfig = {};
-          angular.forEach($scope.builds, function(build, buildName) {
-            var buildConfigName = buildConfigForBuild(build) || "";
-            if (showBuild(build)) {
-              $scope.buildsByBuildConfig[buildConfigName] = $scope.buildsByBuildConfig[buildConfigName] || {};
-              $scope.buildsByBuildConfig[buildConfigName][buildName] = build;
-            }
-          });
-          // Make sure there is an empty hash for every build config we know about
+          $scope.latestByConfig = BuildsService.latestBuildByConfig($scope.builds, showBuild);
+          // Make sure there is a key for every build config we know about
           angular.forEach($scope.buildConfigs, function(buildConfig, buildConfigName){
-            $scope.buildsByBuildConfig[buildConfigName] = $scope.buildsByBuildConfig[buildConfigName] || {};
+            $scope.latestByConfig[buildConfigName] = $scope.latestByConfig[buildConfigName] || null;
           });
         }
 
         function updateFilterWarning() {
-          if (!LabelFilter.getLabelSelector().isEmpty() && $.isEmptyObject($scope.buildsByBuildConfig)) {
+          if (!LabelFilter.getLabelSelector().isEmpty() && $.isEmptyObject($scope.buildConfigs)) {
```

Comment:
buildsByBuildConfig could have had keys for build configs that had been deleted but whose builds still existed, checking if buildConfigs is empty doesn't result in the same answer.  shouldn't this be if latestByConfig is empty?